source/server/http/admin.cc

```diff
@@ -643,20 +658,27 @@ Http::Code AdminImpl::handlerAdminHome(const std::string&, Http::HeaderMap& resp
   response_headers.insertContentType().value().setReference(
       Http::Headers::get().ContentTypeValues.Html);
 
+  response.add(absl::StrReplaceAll(AdminHtmlStart, {{"@FAVICON@", EnvoyFavicon}}));
+
   // Prefix order is used during searching, but for printing do them in alpha order.
-  std::map<std::string, const UrlHandler*> sorted_handlers;
-  for (const UrlHandler& handler : handlers_) {
-    sorted_handlers[handler.prefix_] = &handler;
-  }
+  for (const UrlHandler* handler : sortedHandlers()) {
+    const std::string& url = handler->prefix_;
+
+    // For handlers that mutate state, render the link as a button in a POST form,
+    // rather than an anchor tag. This should discourage crawlers that find the /
+    // page from accidentally mutating all the server state by GETting all the hrefs.
+    const char* link_format =
+        handler->mutates_server_state_
+            //? "<button onclick='window.location.href=\"{}\"'>{}</button>"
```

Comment:
do you mean for this comment to be here?

Reply:
nope, deleting.